src/components/Card/component.tsx

```diff
@@ -1,23 +1,32 @@
 import React from 'react';
 
+import { HtmlTag } from '../../modules/html-tag';
 import { Testable } from '../../modules/test-ids';
 
 import { Container, Position, Variant } from './styled';
 
-export type Props = React.AllHTMLAttributes<HTMLElement> &
+export type Props = Omit<React.AllHTMLAttributes<HTMLElement>, 'as'> &
   Testable & {
+    htmlTag?: HtmlTag;
     position?: Position;
     variant?: Variant;
   };
 
 export const Component = ({
   children,
-  className,
+  htmlTag,
   'data-testid': testId,
   position = 'center',
   variant = 'default',
+  ...otherProps
 }: Props) => (
-  <Container className={className} data-testid={testId} position={position} variant={variant}>
+  <Container
+    {...otherProps}
+    as={htmlTag as any}
```

Comment:
Just curious about `htmlTag as any`. It seems that we use this in some files. Is there any type we can use here? Not sure, maybe `HTMLElementTagNameMap`

Reply:
At the moment we have:
```
export type HtmlTag = NonNullable<keyof React.ReactHTML>;
```

I'm not sure what the difference between this and `HTMLElementTagNameMap` is, I'll have to research or if you recommend we change it we can. Note that even if we use `HTMLElementTagNameMap`, it seems we still need the `as any` cast.

<img width="469" alt="image" src="https://user-images.githubusercontent.com/15721063/98083032-96697a00-1ede-11eb-8085-a6bf9c0889c9.png">

Reply:
Yea we can do this later as a separate task. I am not also 100% sure. My comment here might be redundant :satisfied: